src/pages/Registration.tsx

```
@@ -99,51 +104,70 @@ const Registration: React.FC = () => {
     const [userState, setUserState] = useState<string>('bw');
     const [newsletter, setNewsletter] = useState<boolean>(false);
 
-    const [register] = useMutation(userType === 'pupil' ? mutPupil : mutStudent);
+    const [registerPupil] = useMutation(mutPupil);
+    const [registerStudent] = useMutation(mutStudent);
+
+    const { data: corpData } = useQuery(
+        gql(
+            `query Cooperations {
+            cooperations {
+                id
+                tag
+                name
+                welcomeTitle
+                welcomeMessage
+            }
+        }`
+        )
+    );
+
+    const cooperationTag = new URL(window.location.toString()).searchParams.get('cooperation');
+
+    const cooperation = useMemo(() => {
+        const result = corpData?.cooperations?.find((it) => it.tag === cooperationTag);
+
+        if (cooperationTag && corpData && !result) {
+            throw new Error(`Invalid Cooperation Tag '${cooperationTag}'`);
```

/nit
Should be fine for our fast DHL change, but I think we generally should try to avoid crashing hard

---

Hmmm, I think a misconfiguration of this should rather be detected by us (this shows the issue reporter) than being ignored (and thus we might report wrong numbers)

---

That's true, but failing hard is different than explaining what happened. 
It's the first time the people are interacting with our page and they have no idea if it's down or if it's just a broken link.

Btw. should there be an option to disable the cooperation?

---

Fair enough, I'll build something for that